### PR TITLE
support datetime64

### LINKED ISSUE #3338: DateTime64 DataType support
So we should support enum type~

logical datatype to physical datatype mapping

DateTime64 -> UInt64

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

implement DateTime64 data type

## Changelog

- Improvement

## Related Issues

Fixes #3338

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/functions/src/scalars/expressions/cast.rs [hunk 1]
```diff
@@ -199,6 +214,13 @@ fn string_to_datetime(date_str: impl AsRef<[u8]>) -> Option<NaiveDateTime> {
     s.and_then(|c| NaiveDateTime::parse_from_str(c, "%Y-%m-%d %H:%M:%S").ok())
 }
 
+// TODO support timezone
+#[inline]
+fn string_to_datetime64(date_str: impl AsRef<[u8]>) -> Option<NaiveDateTime> {
```
Comment (sundy-li) [inline common/functions/src/scalars/expressions/cast.rs:219]: Is there any stateless test about datetime64 feature?

https://github.com/datafuselabs/databend/blob/4af0e109e8fba997ed78ad9e3f38b0cb02015f0b/tests/suites/0_stateless/02_0012_function_datetimes.sql

Comment (b41sh) [reply common/functions/src/scalars/expressions/cast.rs:225]: I have added some tests, PTAL @sundy-li

**Review by sundy-li — APPROVED** [review]
/LGTM

**Comment by BohuTANG** [comment]
Thank you @b41sh , great job!

File: common/datavalues/src/types/date_converter.rs [hunk 2]
```diff
@@ -36,4 +37,12 @@ where T: AsPrimitive<i64>
     fn to_date_time(&self, tz: &Tz) -> DateTime<Tz> {
         tz.timestamp_millis(self.as_() * 1000)
     }
+
+    fn to_date_time64(&self, precision: &u32, tz: &Tz) -> DateTime<Tz> {
+        if *precision <= 3 {
```
[the remark this answers is not in the document]
Comment (b41sh) [reply common/datavalues/src/types/date_converter.rs:42]: The range of precision is 0-9, and the typically used values are 3, 6, 9
https://clickhouse.com/docs/en/sql-reference/data-types/datetime64/

**Pull request merged**